python/pyspark/sql/tests/test_python_streaming_datasource.py

```diff
@@ -150,50 +151,90 @@ def check_batch(df, batch_id):
         q.awaitTermination
```

Comment:
@chaoqin-li1123 can we fix all those instances?

Reply:
Could you elaborate? is this test flaky?

Reply:
`q.awaitTermination()`

Reply:
Thanks for reminding!